wip: start writing new index (#346)
plt-amy/1lab: src/index.lagda.md

module index where
```

# 1Lab

This website is an experiment in **discoverable formalisation**: an

> extensive library of formalised mathematics, presented as an explorable

Reviewer: I think it's good that we're being more broad about what the 1Lab formalises, but I do think it's worth mentioning the cubical nature of the formalisation early on, somewhere in the first few sentences?

reference resource. Our implementation of a mathematical library, using
*literate source code*, has allowed us a dual approach to the
explanation of mathematical concepts: the code and the prose are
complementary, and each reader is presented with the opportunity to
choose their own balance of rigid formalisation vs. intuitive
explanation.

<!--
```agda
open import 1Lab.Univalence
open import 1Lab.Equiv
open import 1Lab.HLevel
open import 1Lab.Type
open import 1Lab.Path
open import Cat.Base
_ = Precategory
```
-->

> Making the formalisation explorable requires taking the Web-native

Reviewer: ```suggestion
Making the formalisation explorable requires taking the web-native
```

::: mathpar

```agda
_ : ∀ {ℓ} {A B : Type ℓ} → is-equiv (path→equiv {A = A} {B})
_ = univalence
```

:::

The source code snippet above, a re-statement of the [[univalence
principle|univalence]], exemplifies the hypertext nature of our
presentation. Each reader is allowed to explore the statement, which
might lead them directly to [[type universes|universe]],
[[equivalences]], and [[path types|path]] --- and to further discoveries
from there.

Since the library's size may make jumping into a particular topic feel
daunting, we have done our best to make entry points discoverable as
well. The simplest way to find something is by searching for it. If you
have JavaScript enabled, you can hit <kbd>Ctrl+K</kbd>/<kbd>Cmd+K</kbd>
to bring up the search prompt. Anything that has a name can be searched
for:

- Pages can be looked up by their module name (e.g. _[Algebra.Ring]_),
  or by their title (e.g. _[Abelian groups]_);

- Sections within a page can be looked up by their title, like _[The
  canonical self-indexing > As a fibration][selfindex]_. Keep in mind that
  the fuzzy matching means you don't have to write out the entire

>   prefix; it's only displayed so you know where you'll be taken.

Reviewer: Everywhere else we refer to "the reader", but here uses "you". We should probably use "the reader" here too.

Reviewer 2: I see more "you"s than "reader"s, and I think "you" is more natural.

Author: Both times "reader" shows up it's in the construction _each reader_, though. I don't think there's a change to be made here

- Particular *concepts* can be looked up by any of their shorthands,
  e.g. [[adjunctions compose]] or [[composition of adjunctions]].

- Every definition in the library can be looked up by its identifier,
  like `Precategory`{.Agda}.

[Algebra.Ring]: Algebra.Ring.html
[Abelian groups]: Algebra.Group.Ab.html
[selfindex]: Cat.Displayed.Instances.Slice.html#as-a-fibration

While it is infeasible to maintain a list of everything noteworthy, we
do maintain an association between definitions in the 1Lab and in the
HoTT book: the [HoTT](HoTT.html) page. It's worth mentioning explicitly

> that this page is also indexed for search, so you can find definitions
> by searching for their name in the book.

Reviewer: ```suggestion
that this page is also indexed for search, so you can find a definition
by searching for its name in the book.
```

## Getting involved

Even though the 1Lab is maintained primarily by [the authors][authors],

> it is first and foremost a community project. The literal work takes

Reviewer: Maybe "practical work"? Or "the formalisation"? "literal" is a bit ... messy.

place on [GitHub], but ongoing discussion happens on [Discord]. We
welcome all contributions, though we would kindly ask that anyone
submitting a substantial change discuss it on Discord first.

[GitHub]: https://github.com/plt-amy/1lab
[Discord]: https://discord.gg/eQDNM26uKP
[authors]: Authors.html

Our purpose is to make univalent mathematics accessible to everyone who
is interested, regardless of cultural background, age, educational

> background, ethnicity, gender identity, or gender expression. **We
> believe HoTT is for everyone and are committed to fostering a kind,

Reviewer: "We believe" makes the statement less strong.

```suggestion
background, ethnicity, gender identity, or gender expression. **HoTT
is for everyone and are committed to fostering a kind,
```

  the terms of the Apache License, 2.0; a copy is available
  [here][LICENSE.Noto]. Other icons come from
  [octicons](https://github.com/primer/octicons), which is distributed
  under the MIT license, which you can find
  [here][LICENSE.Octicons].

* **Diagrams**: Our diagrams are created using [quiver], and rendered
  to SVG using [LaTeX] and [pdftocairo], which is part of the
  Poppler project. No part of these projects is redistributed.

[quiver]: https://q.uiver.app
[LaTeX]: https://tug.org/texlive
[pdftocairo]: https://poppler.freedesktop.org
[Pandoc]: https://pandoc.org

[Julia Mono]: https://juliamono.netlify.app/
[Inria Sans]: https://github.com/BlackFoundryCom/InriaFonts
[EB Garamond]: https://github.com/georgd/EB-Garamond
[OFL]: https://openfontlicense.org/

[LICENSE.JuliaMono]: static/licenses/LICENSE.JuliaMono.txt
[LICENSE.InriaSans]: static/licenses/LICENSE.InriaSans.txt
[LICENSE.EBGaramond]: static/licenses/LICENSE.EBGaramond.txt
[LICENSE.Noto]: static/licenses/LICENSE.Noto.txt
[LICENSE.Octicons]: static/licenses/LICENSE.Octicons.txt
[LICENSE.KaTeX]: static/licenses/LICENSE.KaTeX.txt

[KaTeX]: https://katex.org

## Recently added files

> ::: {#recent-additions}

Reviewer: Oooooh, this is snazzy.
